### PR TITLE
introduced force_2d for a subset of geometries #180

### PR DESCRIPTION
**Pull request by whisk**
Added `force_2d` function for Point, MultiPoint, LineString, Polygon and all descending classes (if any). No support for MultiLineString and MultiPolygon yet.

### CODE REVIEW

**Comment by cleder** [comment]
Hi @whisk, thanks for your contribution.

I think that the `force_2d` and `force_3d` could be cleaner to implement by operating on the `__geo_interface__`  like [pygeoif.functions.compare_geo_interface](https://github.com/cleder/pygeoif/blob/8d349699af24931c7e465328d6c11bd0b98187af/pygeoif/functions.py#L166)

`def force_2d(geometry: Union[GeoType, GeoCollectionType]) -> Union[Geometry, GeometryCollection]:`

For `force_2d` you can [zip](https://docs.python.org/3.3/library/functions.html#zip) the coordinates with `(0 ,0)`, 

`def force_3d(geometry: Union[GeoType, GeoCollectionType], z: float = 0) -> Union[Geometry, GeometryCollection]:`

For `force_3d` you could use [zip_longest](https://docs.python.org/3/library/itertools.html#itertools.zip_longest) using `z` as the `fillvalue`

To convert the resulting `__geo_interface__` dictionary back into a geometry, you may use [shape](https://github.com/cleder/pygeoif/blob/8d349699af24931c7e465328d6c11bd0b98187af/pygeoif/factories.py#L100)

I hope this makes sense.

**Comment by cleder** [comment]
https://realpython.com/python-zip-function/ and https://ioflood.com/blog/python-zip-function/ seem to be useful to learn more about `zip`

**Comment by cleder** [comment]
Approach I'd take:

```
def move_coordinate(coordinate: Sequence[float], move_by: Sequence[float]) -> Tuple[float, ...]:
    if len(coordinate) > len(move_by):
        return tuple(c + m for c, m in zip(coordinate, move_by))
    else:
        return tuple(c + m for c, m in zip_longest(coordinate, move_by, fillvalue=0.0))
```
```
>>> move_coordinate((0,0), (0,0,0))
(0, 0, 0.0)
>>> move_coordinate((0,0), (0,0,1))
(0, 0, 1.0)
>>> move_coordinate((0,0,0), (0,0,1))
(0, 0, 1)
>>> move_coordinate((0,0,0), (0,0))
(0, 0)
```
This can be used by force_2d with `(0, 0)` as `move_by` and force_3d with `(0, 0, z)` as move_by

**Comment by whisk** [comment]
Hi, I refactored `force_2d` by using `__geo_interface__`, like functions `compare_geo_interface` and `shape` do. This is much simpler and I think that's what you were looking for.

I also added `move_coordinate` and `move_coordinates` functions.

File: pygeoif/functions.py [hunk 1]
```diff
@@ -188,11 +190,60 @@ def compare_geo_interface(
...
+    This forcefully changes the dimension of each of the coordinate to match
+    the dimension of the vector.
+    >>> move_coordinates(((0, 0), (-1, 1)), (-1, 1))
+    ((-1, 1), (-2, 2))
+    >>> move_coordinates(((0, 0, 0), (-1, 1, 0)), (-1, 1))
+    ((-1, 1), (-2, 2))
+    >>> move_coordinates(((0, 0), (-1, 1)), (-1, 1, 0))
+    ((-1, 1, 0), (-2, 2, 0))
+    """
+    if isinstance(coordinates, (tuple, list)) and isinstance(
```
Comment (whisk) [inline pygeoif/functions.py:231]: I preferred to check types explicitly rather what catch TypeError exception.

**Review by cleder — APPROVED** [review]
Nice work :-) Thanks a lot!

File: pygeoif/functions.py (hunk 1 above)
Comment (whisk) [reply pygeoif/functions.py]: Thanks for pointing this out!
As far as I see, you refactored this lines to use `CoordinatesType`, not `Sequence[Any]`, so this function just can't accept generators anyway. 
Not sure if generators still need to be supported in `move_coordinates`?

Comment (cleder) [reply pygeoif/functions.py]: You still CAN pass in a generator, it is just that mypy will complain that you should not do that

Comment (cleder) [reply pygeoif/functions.py]: Thanks again for your help, your work will be in the next release (1.2) :+1:

Comment (whisk) [reply pygeoif/functions.py]: > You still CAN pass in a generator, it is just that mypy will complain that you should not do that

Yes, yes, you are right. I'm probably just too used to strict typing :)
Please consider PR #188 regarding it.

**Pull request merged**